test/test_puma_server.rb

```diff
@@ -189,6 +189,36 @@ def test_early_hints_works
     assert_equal expected_data, data
   end
 
+  def test_early_hints_are_ignored_if_connection_lost
+    # Have a long early hints header so we have time to close connection
+    # before the server sends the whole content.
+    links = Array.new(100) { |n| "</script_#{n}.js>; rel=preload" }.join("\n")
```

Comment:
Rather than do this, which is a nondeterministic test, let's just do something like:

```ruby
server = Puma::Server.new app, events
def server.fast_write(*args)
  raise ConnectionError
end
```

Reply:
Ah, great! I wanted to do that but I couldn't figure out how to modify `#fast_write` without affecting the whole test suite.

Thanks!